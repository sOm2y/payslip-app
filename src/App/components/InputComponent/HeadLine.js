//@flow

import React from 'react';
import Typography from 'material-ui/Typography';

class HeadLine extends React.PureComponent<{}> {
  render() {
    return (
      <div
        style={{
          height: 50
        }}
      >
        <Typography type="headline" id="title">
          Please input your detail
        </Typography>
      </div>
    );
  }
}

export default HeadLine;
