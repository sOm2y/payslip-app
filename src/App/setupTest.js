//@flow

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// enable using of expect(wrapper).toContainReact(<SomeComponent/>)
import 'jest-enzyme';

configure({ adapter: new Adapter() });
