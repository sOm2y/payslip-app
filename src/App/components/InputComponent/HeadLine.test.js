//@flow

import React from 'react';
import HeadLine from './HeadLine';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import '../../setupTest'; // enzyme setup

describe('Actions', () => {
  it('should renders without crashing', () => {
    shallow(<HeadLine />);
  });

  it('should contain the title', () => {
    const headline = shallow(<HeadLine />);
    expect(headline.html()).toContain('Please input your detail');
  });

  it('should match snapshot', () => {
    const tree = renderer.create(<HeadLine />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
