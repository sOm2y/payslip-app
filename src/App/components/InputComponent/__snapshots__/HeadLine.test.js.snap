// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions should match snapshot 1`] = `
<div
  style={
    Object {
      "height": 50,
    }
  }
>
  <h1
    className="MuiTypography-root-1 MuiTypography-headline-6"
    id="title"
  >
    Please input your detail
  </h1>
</div>
`;
